apps/src/assetManagement/animationLibraryApi.js

```diff
@@ -29,9 +34,37 @@ export function uploadSpriteToAnimationLibrary(
     body: imageData
   })
     .then(response => {
-      onSuccess(response);
+      onSuccess(UploadType.SPRITE, response);
+    })
+    .catch(err => {
+      onError(UploadType.SPRITE, err);
+    });
+}
+
+/* Uploads the given JSON of sprite metadata to the animation library at the specified path. On success
+  and error calls the associated function
+ * @param destination {String} path to metadata location in the animation-library folder
+ * @param jsonData {String} JSON object of metadata to upload
+ * @param onSuccess {function} callback function for success upload
+ * @param onError {function} callback function for upload error
+ */
+export function uploadMetadataToAnimationLibrary(
+  destination,
+  jsonData,
+  onSuccess,
+  onError
+) {
+  return fetch(`/api/v1/animation-library` + destination, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    body: jsonData
+  })
+    .then(response => {
+      onSuccess(UploadType.METADATA, response);
```

Comment:
moving [this thread](https://github.com/code-dot-org/code-dot-org/pull/40666/files#r638225127) here. maybe i'm misunderstanding what you're saying, but i think we can handle all 3 states if we refactor these API methods in the following way:

```javascript
fetch(url, {})
  .then(response => {
    response.ok ? onSuccess() : onError()
  }).catch(err => {
    onError()
  })
```

i think that will cover all 3 states you mentioned: success (aka response.ok), !response.ok, and promise rejected (in catch)

Reply:
Nope, that is correct, I just needed to change the onSuccess and onError parameters and I had been following a different road with the promises work on Friday, so I was getting myself turned aroun. I ended up doing `response.ok? onSuccess : throw new Error` and porting everything through the catch and handling the error/rejecting the promise there. I'm currently working on that commit and it should be up in a few minutes.